.test-infra/metrics/sync/github/sync.py

```diff
@@ -322,63 +379,107 @@ def upsertIntoPRsTable(cursor, values):
+    connection.close()
+    if lastSyncTimestamp is None:
+      if kind == 'pr':
+        connection = initDBConnection()
+        cursor = connection.cursor()
+        lastSyncTimestamp = fetchLastSyncTimestampFallback(cursor)
+        cursor.close()
+        connection.close()
+      else:
+        lastSyncTimestamp = datetime(year=1980, month=1, day=1)
+
+    currTS = lastSyncTimestamp
+
+    resultsPresent = True
+    while resultsPresent:
+      print(f'Syncing data for {kind}s: ', currTS)
+      query = queries.MAIN_PR_QUERY if kind == 'pr' else queries.MAIN_ISSUES_QUERY
+      jsonData = fetchGHData(currTS, query)
+
+      connection = initDBConnection()
+      cursor = connection.cursor()
+
+      if "errors" in jsonData:
+        print("Failed to fetch data, error:", jsonData)
+        return
 
-    if "errors" in jsonData:
-      print("Failed to fetch data, error:", jsonData)
-      return
-
-    prs = None
-    try:
-      prs = jsonData["data"]["search"]["edges"]
-    except:
-      # TODO This means that API returned error.
-      # We might want to bring this to stderr or utilize other means of logging.
-      # Examples: we hit throttling, etc
-      print("Got bad json format: ", jsonData)
-      return
-
-    if not prs:
-      resultsPresent = False
-
-    for edge in prs:
-      pr = edge["node"]
+      data = None
       try:
-        rowValues = extractRowValuesFromPr(pr)
-      except Exception as e:
-        print("Failed to extract data. Exception: ", e, " PR: ", edge)
-        traceback.print_tb(e.__traceback__)
+        data = jsonData["data"]["search"]["edges"]
+      except:
+        # TODO This means that API returned error.
+        # We might want to bring this to stderr or utilize other means of logging.
+        # Examples: we hit throttling, etc
+        print("Got bad json format: ", jsonData)
         return
 
-      upsertIntoPRsTable(cursor, rowValues)
+      if not data:
+        resultsPresent = False
 
-      prUpdateTime = ghutilities.datetimeFromGHTimeStr(pr["updatedAt"])
+      for edge in data:
+        node = edge["node"]
+        try:
+          rowValues = extractRowValuesFromPr(node) if kind == 'pr' else extractRowValuesFromIssue(node)
+        except Exception as e:
+          print("Failed to extract data. Exception: ", e, f" {kind}: ", edge)
+          traceback.print_tb(e.__traceback__)
+          return
 
-      currTS = currTS if currTS > prUpdateTime else prUpdateTime
+        if kind == 'pr':
+          upsertIntoPRsTable(cursor, rowValues)
+        else:
+          upsertIntoIssuesTable(cursor, rowValues)
 
-    cursor.close()
-    connection.commit()
-    connection.close()
+        updateTime = ghutilities.datetimeFromGHTimeStr(node["updatedAt"])
+
+        currTS = currTS if currTS > updateTime else updateTime
+
+      cursor.close()
+      connection.commit()
+      connection.close()
 
-    updateLastSyncTimestamp(currTS)
+      updateLastSyncTimestamp(currTS, 'gh_pr_sync')
```

Comment:
Should this be `gh_{kind}_sync` as well?

Reply:
Yes, that's a good catch. Updated!